lib/oeqa/selftest/cases/updater_qemux86_64_ptest.py

```diff
@@ -0,0 +1,52 @@
+# pylint: disable=C0111,C0325
+import os
+import re
+
+from oeqa.selftest.case import OESelftestTestCase
+from oeqa.utils.commands import runCmd
+from testutils import qemu_launch, qemu_send_command, qemu_terminate
+
+
+class PtestTests(OESelftestTestCase):
+
+    def setUpLocal(self):
+        layer = "meta-updater-qemux86-64"
+        result = runCmd('bitbake-layers show-layers')
+        if re.search(layer, result.output) is None:
+            # Assume the directory layout for finding other layers. We could also
+            # make assumptions by using 'show-layers', but either way, if the
+            # layers we need aren't where we expect them, we are out of like.
+            path = os.path.abspath(os.path.dirname(__file__))
+            metadir = path + "/../../../../../"
+            self.meta_qemu = metadir + layer
+            runCmd('bitbake-layers add-layer "%s"' % self.meta_qemu)
+        else:
+            self.meta_qemu = None
+        self.append_config('MACHINE = "qemux86-64"')
+        self.append_config('SYSTEMD_AUTO_ENABLE_aktualizr = "disable"')
+        self.append_config('PTEST_ENABLED_pn-aktualizr = "1"')
+        self.append_config('IMAGE_INSTALL_append += "aktualizr-ptest ptest-runner "')
+        self.qemu, self.s = qemu_launch(machine='qemux86-64')
+
+    def tearDownLocal(self):
+        qemu_terminate(self.s)
+        if self.meta_qemu:
+            runCmd('bitbake-layers remove-layer "%s"' % self.meta_qemu, ignore_status=True)
+
+    def qemu_command(self, command, timeout=60):
+        return qemu_send_command(self.qemu.ssh_port, command, timeout=timeout)
+
+    def test_run_ptests(self):
+        # simulate a login shell, so that /usr/sbin is in $PATH (from /etc/profile)
+        stdout, stderr, retcode = self.qemu_command('sh -l -c ptest-runner', timeout=None)
+        output = stdout.decode()
+        print(output)
+        self.assertEqual(retcode, 0)
```

Comment:
The log not showing might be cause by this assert breaking. I'll try to see if moving it helps

Reply:
Hm actually it looks like a timeout triggered by ptest-runner when I tried it. And ctest first dumps its output in `/tmp/aktualizr-ptest.log.tmp` before renaming it to its final name, presumably at the end.

So, if ctest is killed, it would still keep the old name.

Reply:
Yeah, I've been seeing that timeout, too. Somehow it isn't a problem on master/thud.

Reply:
In my case, the only test that doesn't terminate is `test_fetcher`.

Reply:
```
Traceback (most recent call last):
  File "tests/sota_tools/treehub_server.py", line 14, in <module>
    from pathlib import Path
ImportError: No module named 'pathlib'
```

That's a bit annoying that the test doesn't terminate after that... But at least it should be fixable.